### ISSUE TITLE
Create a bit set API

### ISSUE THREAD

**Issue by singhbabneet** [eclipse/omr#1162]
This API will be part of the port library. It will help reduce redundant code in the port library.

```
typedef OMR_Bitfield uint32_t*; 

omrutil_bitfield_is_set(const OMR_Bitfield bitfild, size_t index); 

omrutil_bitfield_set(OMR_Bitfield bitfield, size_t index);
```

**Comment by 0xdaryl** [comment]
Note that the OMR compiler already has something similar (see `compiler/infra/Flags.hpp`).  Perhaps there's an opportunity for a similar solution to be used across multiple components in OMR (I'm not saying its Flags.hpp or what you are proposing, but perhaps some middle ground that can be used everywhere).  Please keep it in mind.

**Comment by singhbabneet** [comment]
Agreed. Flags.hpp has a lot of functionality which can be used to create a bit manipulation API. But, port library won't be able to support C++ code. DDR will not function properly. So, an equivalent C API can be added. `template <class T>`: `T` can be replaced with `uint8_t*` or `uint8_t[SIZE]` to support different flag sizes.
